core/src/executor/evaluate/mod.rs

```diff
@@ -204,407 +199,156 @@ async fn evaluate_function<'a, T: Debug>(
     aggregated: Option<Rc<HashMap<&'a Aggregate, Value>>>,
     func: &'a Function,
 ) -> Result<Evaluated<'a>> {
+    use function as f;
+
     let eval = |expr| {
         let context = context.as_ref().map(Rc::clone);
         let aggregated = aggregated.as_ref().map(Rc::clone);
 
         evaluate(storage, context, aggregated, expr)
     };
 
-    enum Nullable<T> {
-        Value(T),
-        Null,
-    }
-
-    let eval_to_str = |expr| async move {
-        match eval(expr).await?.try_into()? {
-            Value::Str(s) => Ok(Nullable::Value(s)),
-            Value::Null => Ok(Nullable::Null),
-            _ => {
-                Err::<_, Error>(EvaluateError::FunctionRequiresStringValue(func.to_string()).into())
-            }
-        }
-    };
-
-    let eval_to_float = |expr| async move {
-        match eval(expr).await?.try_into()? {
-            Value::I64(v) => Ok(Nullable::Value(v as f64)),
-            Value::F64(v) => Ok(Nullable::Value(v)),
-            Value::Null => Ok(Nullable::Null),
-            _ => {
-                Err::<_, Error>(EvaluateError::FunctionRequiresFloatValue(func.to_string()).into())
-            }
-        }
-    };
-
-    let eval_to_integer = |expr| async move {
-        match eval(expr).await?.try_into()? {
-            Value::I64(number) => Ok(Nullable::Value(number)),
-            Value::Null => Ok(Nullable::Null),
-            _ => Err::<_, Error>(
-                EvaluateError::FunctionRequiresIntegerValue(func.to_string()).into(),
-            ),
-        }
-    };
-
-    macro_rules! eval_to_str {
-        ($v : expr) => {
-            match eval_to_str($v).await? {
-                Nullable::Value(s) => s,
-                Nullable::Null => return Ok(Evaluated::from(Value::Null)),
-            }
-        };
-    }
-
-    macro_rules! eval_to_float {
-        ($v : expr) => {
-            match eval_to_float($v).await? {
-                Nullable::Value(f) => f,
-                Nullable::Null => return Ok(Evaluated::from(Value::Null)),
-            }
-        };
-    }
-
-    macro_rules! eval_to_integer {
-        ($v : expr) => {
-            match eval_to_integer($v).await? {
-                Nullable::Value(i) => i,
-                Nullable::Null => return Ok(Evaluated::from(Value::Null)),
-            }
-        };
-    }
+    let name = || func.to_string();
```

Comment:
Since there is no place to take the Name after the match expression, there seems to be no ownership problem.
How about converting from closure type to String?

```suggestion
    let name = func.to_string();
```

Reply:
This is because.. there are some functions which does not require `name`.

e.g.
```rust
Function::GenerateUuid() => Ok(f::generate_uuid()),
Function::Now() => Ok(Value::Timestamp(Utc::now().naive_utc())),
```